File: docs/vs-2015/snippets/csharp/VS_Snippets_VSSDK/vssdkmenucommandtest/cs/commentadornmenttest/commentadornment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
//<Snippet1>
using Microsoft.VisualStudio.Text;
//</Snippet1>

namespace CommentAdornmentTest
{
    //<Snippet2>
    internal class CommentAdornment
    //</Snippet2>
    {
        //<Snippet3>
        public readonly ITrackingSpan Span;
        public readonly string Author;
        public readonly string Text;
        //</Snippet3>

        //<Snippet4>
        public CommentAdornment(SnapshotSpan span, string author, string text)
        {
            this.Span = span.Snapshot.CreateTrackingSpan(span, SpanTrackingMode.EdgeExclusive);
            this.Author = author;
            this.Text = text;
        }
        //</Snippet4>
    }
}

File: docs/vs-2015/snippets/visualbasic/VS_Snippets_VSSDK/vssdkmenucommandtest/vb/commentadornmenttest/commentadornment.vb
'<Snippet1>
Imports Microsoft.VisualStudio.Text
'</Snippet1>

'<Snippet2>
Friend Class CommentAdornment
    '</Snippet2>

    '<Snippet3>
    Public ReadOnly Span As ITrackingSpan
    Public ReadOnly Author As String
    Public ReadOnly Text As String
    '</Snippet3>

    '<Snippet4>
    Public Sub New(ByVal span As SnapshotSpan, ByVal author As String, ByVal text As String)
        Me.Span = span.Snapshot.CreateTrackingSpan(span, SpanTrackingMode.EdgeExclusive)
        Me.Author = author
        Me.Text = text
    End Sub
    '</Snippet4>
End Class